src/pages-helpers/curriculum/xlsx/helper.test.ts

```diff
@@ -192,3 +202,294 @@ describe("addOrUpdateSheet", () => {
+      phaseSlug: "secondary",
+      ks4OptionSlug: "gcse",
+      keyStageSlug: undefined,
+      tierSlug: "foundation",
+      childSubjectSlug: "chemistry",
+    };
+    const output = generateYearTitle(formattedData, year, slugs);
+    expect(output).toEqual("Year 11 Chemistry, Foundation");
+  });
+
+  test("Year 10 Combined science, Higher", () => {
+    const formattedData = {
+      yearData: {
+        "11": createYearData({
+          units: [
+            createUnit({
+              year: "11",
+              subject_slug: "chemistry",
+              examboard: "AQA",
+              tier: "Foundation",
+            }),
+          ],
+        }),
+      },
+      threadOptions: [],
+      tiers: ["foundation"],
+      yearOptions: ["11"],
+    };
+    const year = "11";
+    const slugs = {
+      subjectSlug: "science",
+      phaseSlug: "secondary",
+      ks4OptionSlug: "gcse",
+      keyStageSlug: undefined,
+      tierSlug: "higher",
+      childSubjectSlug: "chemistry",
+    };
+    const output = generateYearTitle(formattedData, year, slugs);
+    expect(output).toEqual("Year 11 Chemistry, Higher");
+  });
+
+  // When a year has a subject title override (i.e. KS4 GCSE Computing = Computer science), update the subject name accordingly in the sheet title.
+  test("Year 10 Computer science, GCSE", () => {
+    const formattedData = {
+      yearData: {
+        "10": createYearData({
+          units: [
+            createUnit({
+              year: "10",
+              subject_slug: "computing",
+              examboard: "AQA",
+            }),
+          ],
+        }),
+      },
+      threadOptions: [],
+      tiers: ["foundation"],
+      yearOptions: ["10"],
+    };
+    const year = "10";
+    const slugs = {
+      subjectSlug: "computing",
+      phaseSlug: "secondary",
+      ks4OptionSlug: "aqa",
+    };
+    const output = generateYearTitle(formattedData, year, slugs);
+    expect(output).toEqual("Year 10 Computer Science, GCSE");
+  });
+
+  // Year groups that do not have these modifiers (i.e. Years 1-9) should have default titles
+  test("Year 9 Physical education", () => {
+    const formattedData = {
+      yearData: {
+        "9": createYearData({
+          units: [
+            createUnit({
+              year: "9",
+              subject: "Physical education",
+            }),
+          ],
+        }),
+      },
+      threadOptions: [],
+      yearOptions: ["9"],
+    };
+    const year = "9";
+    const slugs = {
+      subjectSlug: "physical-education",
+      phaseSlug: "secondary",
+      ks4OptionSlug: undefined,
+      keyStageSlug: undefined,
+      tierSlug: undefined,
+      childSubjectSlug: undefined,
+    };
+    const output = generateYearTitle(formattedData, year, slugs);
+    expect(output).toEqual("Year 9 Physical education");
+  });
+
+  // When units are grouped as all-years, replace the sheet title with {groupAs} (all years) and the worksheet name with {groupAs}
```

Comment:
The comment suggests that the worksheet name is being tested as well as the sheet title, but there's no actual test steps to confirm this.